packages/flutter/lib/src/painting/gradient.dart

```diff
@@ -606,7 +606,7 @@ class RadialGradient extends Gradient {
       radius * rect.shortestSide,
       colors, _impliedStops(), tileMode,
       null, // transform
-      focal == null  ? null : focal.resolve(textDirection).withinRect(rect),
+      focal?.resolve(textDirection)?.withinRect(rect),
```

Comment:
ditto

Reply:
i doubt this has much effect but it's probably worth benchmarking one of these cases to check.